pkgs/build-support/node/fetch-npm-deps/src/main.rs

```diff
@@ -31,18 +31,36 @@ struct OldPackage {
     dependencies: Option<HashMap<String, OldPackage>>,
 }
 
-#[derive(Deserialize)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
```

Comment:
Nit: [Eq](https://doc.rust-lang.org/std/cmp/trait.Eq.html) implies (and extends) PartialEq, so PartialEq is redundant here.

Reply:
Not true, `PartialEq` must be implemented to implement `Eq`:

```
can't compare `UrlOrString` with `UrlOrString`
the trait `std::cmp::PartialEq` is not implemented for `UrlOrString`
```